### DISCUSSION TITLE
How to use multi-line input prompt to add several tasks at once ?

### DISCUSSION THREAD

**Discussion by Fred-Vatin** [chhoumann/quickadd#689] [Q&A]
Obsidian | QuickAdd
------- | -------
1.5.12  | v 1.8.1

Hi,

I’m trying to add multiple tasks in a file at once using multi-line prompt, each task being on its own line.

I use capture with task enabled but all lines from my input are added to the same task instead of inserting `- [ ]` before each line.

Any idea how to do that ? 

Bonus: Possible to write a script to check if a task already exist and only add the non-existent ones ? Where could I start ?

**Comment by chhoumann** [comment]
Sorry for the long silence on this one.

The Task toggle wraps the WHOLE capture once, so multi-line input turns into a single `- [ ]` item. To get one task per line, use a small Macro instead - a user script collects the lines and builds the checklist, then a Capture writes it with the Task toggle OFF. As a bonus, the same script skips tasks that are already in the note, which covers your second question too.

Setup:

1. Save this as a user script in your vault, e.g. `scripts/collect-tasks.js` (set `targetPath` to the note you're adding tasks to):

```js
module.exports = async (params) => {
	const { app, quickAddApi } = params;
	const targetPath = "Tasks.md";
	const raw = await quickAddApi.wideInputPrompt("Enter tasks (one per line)");
	const lines = (raw ?? "")
		.split("\n")
		.map((l) => l.trim())
		.filter(Boolean);
	if (lines.length === 0) {
		params.abort("No tasks entered.");
		return;
	}
	const file = app.vault.getAbstractFileByPath(targetPath);
	const existing = file ? await app.vault.read(file) : "";
	const existingSet = new Set(existing.split("\n").map((l) => l.trim()));
	const fresh = lines.filter((l) => !existingSet.has(`- [ ] ${l}`));
	if (fresh.length === 0) {
		params.abort("All of those tasks are already in the note.");
		return;
	}
	params.variables.tasks = fresh.map((l) => `- [ ] ${l}`).join("\n");
};
```

2. Create a Macro ("Add tasks") with two steps, in order:
   - **User Script** -> `collect-tasks.js`
   - **Capture** -> configured as below

3. Capture step:
   - Capture To: your tasks note (the same file as `targetPath` above)
   - Insert after a heading, or bottom of file - wherever you keep them
   - Capture format: `{{VALUE:tasks}}`
   - Task toggle: **OFF** (the script already put `- [ ]` on each line)

Running it opens a multi-line box; type one task per line. I tested it end to end - entering:

```
Buy milk
Call dentist
Existing task

  Water plants  
```

into a note that already had `- [ ] Existing task` produced three new checkboxes (Buy milk, Call dentist, Water plants), skipped the duplicate, dropped the blank line, and trimmed the stray spaces. And if you cancel the box with Esc, the macro stops cleanly and nothing is written.

If you'd like a native "one task per line" capture option, that's a fair feature request to open - but the macro above does the job.

Docs: https://quickadd.obsidian.guide/docs/QuickAddAPI
